### PR TITLE
Avoid deflating layout slot by border thickness twice

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/FrameworkElementExtensions.cs [hunk 1]
```diff
@@ -324,6 +324,48 @@ internal static bool TrySetPadding(this IFrameworkElement frameworkElement, Thic
...
+			{
+				borderThickness = rp.BorderThickness;
+				return true;
+			}
+
+			if (__LinkerHints.Is_Microsoft_UI_Xaml_Controls_LayoutPanel_Available && frameworkElement is Microsoft.UI.Xaml.Controls.LayoutPanel lp)
+			{
+				borderThickness = lp.BorderThickness;
+				return true;
+			}
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/FrameworkElementExtensions.cs:351]: Can all these be replaced by the base `Panel` and use https://github.com/unoplatform/uno/blob/0190ea92ffc138e1cc3259e59fdf2074cbc16465/src/Uno.UI/UI/Xaml/Controls/Panel/Panel.cs#L113 ?

File: src/Uno.UI/UI/Xaml/FrameworkElementExtensions.cs [hunk 2]
```diff
@@ -324,6 +324,48 @@ internal static bool TrySetPadding(this IFrameworkElement frameworkElement, Thic
...
+
+			if (__LinkerHints.Is_Windows_UI_Xaml_Controls_Border_Available && frameworkElement is Border b)
+			{
+				borderThickness = b.BorderThickness;
+				return true;
+			}
+
+			borderThickness = default;
+			return false;
+		}
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/FrameworkElementExtensions.cs:367]: Is this missing `Control.BorderThickness`?

https://github.com/unoplatform/uno/blob/0190ea92ffc138e1cc3259e59fdf2074cbc16465/src/Uno.UI/UI/Xaml/Controls/Control/Control.cs#L630

File: src/Uno.UI/UI/Xaml/FrameworkElementExtensions.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/FrameworkElementExtensions.cs]: Good idea, that simplifies things a lot

File: src/Uno.UI/UI/Xaml/FrameworkElementExtensions.cs (hunk 2 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/FrameworkElementExtensions.cs:349]: In this case it is intentional - only for `Panels`, `ContentPresenter` and `Border` the `BorderThickness` is actually "meaningful". `Control.BorderThickness` does nothing unless it is applied on something within its control template.

**Review by jeromelaban — APPROVED** [review]
Screenshots are not showing anything specific, good :)

**Pull request merged**